tests/testsuite/bad_config.rs

```diff
@@ -350,10 +347,31 @@ fn bad_git_dependency() {
         .file("src/lib.rs", "")
         .build();
 
-    p.cargo("build -v")
-        .with_status(101)
-        .with_stderr(
-            "\
+    let expected_stderr = if uses_gitoxide() {
+        "\
+[UPDATING] git repository `file:///`
+[ERROR] failed to get `foo` as a dependency of package `foo v0.0.0 [..]`
+
+Caused by:
+  failed to load source for dependency `foo`
+
+Caused by:
+  Unable to update file:///
+
+Caused by:
+  failed to clone into: [..]
+
+Caused by:
+  Could not verify that \"file:///\" url [..]
+
+Caused by:
+  Could not retrieve metadata of \"/.git\"
+
+Caused by:
+[..]
```

Comment:
Looking at the UX of this,
- Of the non `[..]` text, this is less clear what is wrong
- The long chain of causes makes this even harder to find what is going on

Reply:
The problem is that neither of these URLs are invalid according to `git`, and `gitoxide` does not want to be more restrictive unless it's fixing a bug (that is not widely used).

* `file:..` is interpreted as SCP-like ssh url by git. It's (somehow) converted into…
* `file:///` though which indicates a repository at the file root `/`

Probably `gitoxide`'s error chain can be made clearer as it's long-windedly saying that it can't read a repository at `/`, but doing that seems to be the correct course of action here.

I can imagine to change the URL so that it is invalid for both implementations, but then `file:..` wouldn't be tested anymore even though it's [based on a bug report](https://github.com/rust-lang/cargo/issues/1884). Since that is also tested [over at `rust-url](https://github.com/servo/rust-url/pull/125/files) I think it's OK to change the URL.

What do you think?

Reply:
To be clear, I do not expect errors to be 100% the same just that the user's experience should be on par.  It would be nice if the chain from gix was shorter and more clear about what happened.

Reply:
Thank you, I think I got it now. 9c4b8cf69ccf45f8f071247df6ff1f485219d411 adjusts the test to compare apples to apples, which does indeed yield the same result as

* the error chain is short
* it tells quite clearly what's wrong. *(I wasn't too happy with the message, but that will be fixed with the next `gix` release)*.